SOURCEBUILD.md

```diff
@@ -0,0 +1,237 @@
+
+### Build and Install the C/C++ CUDA components
+
+To install cuGraph from source, ensure the dependencies are met and follow the steps below:
+
+1) A cloned version of the repository
+
+  ```bash
+  # Set the localtion to cuGraph in an environment variable CUGRAPH_HOME
+  export CUGRAPH_HOME=$(pwd)/cugraph
+
+  # Download the cuGraph repo - if you have a folked version, use that path here instead
+  git clone https://github.com/rapidsai/cugraph.git $CUGRAPH_HOME
+
+  cd $CUGRAPH_HOME
+  ```
+
+2) Create the conda development environment
+
+```bash
+# create the conda environment (assuming in base `cugraph` directory)
+
+# for CUDA 10
+conda env create --name cugraph_dev --file conda/environments/cugraph_dev_cuda10.0.yml
+
+# for CUDA 10.1
+conda env create --name cugraph_dev --file conda/environments/cugraph_dev_cuda10.1.yml
+
+# for CUDA 10.2
+conda env create --name cugraph_dev --file conda/environments/cugraph_dev_cuda10.2.yml
+
+# activate the environment
+conda activate cugraph_dev
+
+# to deactivate an environment
+conda deactivate
+```
+
+  - The environment can be updated as development includes/changes the dependencies. To do so, run:
+
+
+```bash
+
+# for CUDA 10
+conda env update --name cugraph_dev --file conda/environments/cugraph_dev_cuda10.0.yml
+
+# for CUDA 10.1
+conda env update --name cugraph_dev --file conda/environments/cugraph_dev_cuda10.1.yml
+
+# for CUDA 10.2
+conda env update --name cugraph_dev --file conda/environments/cugraph_dev_cuda10.2.yml
+
+conda activate cugraph_dev
+```
+
+3) Build and install `libcugraph`. CMake depends on the `nvcc` executable being on your path or defined in `$CUDACXX`.
+
+  This project uses cmake for building the C/C++ library. CMake will also automatically build and install nvGraph library (`$CUGRAPH_HOME/cpp/nvgraph`) which may take a few minutes. To configure cmake, run:
+
+  ```bash
+  # Set the localtion to cuGraph in an environment variable CUGRAPH_HOME
+  export CUGRAPH_HOME=$(pwd)/cugraph
+
+  cd $CUGRAPH_HOME
+  cd cpp                                        # enter cpp directory
+  mkdir build                                   # create build directory
+  cd build                                      # enter the build directory
+  cmake .. -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX
+
+  # now build the code
+  make -j                                       # "-j" starts multiple threads
+  make install                                  # install the libraries
+  ```
+The default installation locations are `$CMAKE_INSTALL_PREFIX/lib` and `$CMAKE_INSTALL_PREFIX/include/cugraph` respectively.
+
+As a convenience, a `build.sh` script is provided in `$CUGRAPH_HOME`. To execute the same build commands above, run the script as shown below.  Note that the libraries will be installed to the location set in `$PREFIX` if set (i.e. `export PREFIX=/install/path`), otherwise to `$CONDA_PREFIX`.
```

Comment:
`build.sh` does pretty much everything when it comes to building from sources. It is a bit weird to have it in the middle if the step by step instructions. I wonder if it should go first.